common/vector.hpp

```diff
@@ -34,15 +34,64 @@ namespace acommon
       this->resize(pos + s);
       return pos;
     }
-    T * data() {return &*this->begin();}
-    T * data(int pos) {return &*this->begin() + pos;}
-    T * data_end() {return &*this->begin() + this->size();}
+    T * data()
+    {
+      #if __cplusplus >= 201103L
+        return this->data();
+      #else
+        return &*this->begin();
+      #endif
+    }
+    T * data(int pos)
+    {
+      #if __cplusplus >= 201103L
+        return this->data() + pos;
+      #else
+        return &*this->begin() + pos;
+      #endif
+    }
```

Comment:
Please just make this:
```
T * data(int pos) {return data() + pos;}
```

Reply:
Hi,
I'm extremely sorry about that; this was quite clumsy of me to overlook the function signature.
I will make the changes and submit another pull request.

Thanks